Sources/System/FilePath/FilePathParsing.swift

```diff
@@ -89,6 +89,11 @@ extension SystemString {
       // `_prenormalizeWindowsRoots` and resume.
       readIdx = _prenormalizeWindowsRoots()
       writeIdx = readIdx
+
+      // Skip redundant separators
+      while readIdx < endIndex && isSeparator(self[readIdx]) {
+          self.formIndex(after: &readIdx)
+      }
```

Comment:
This is kinda tricky, the duplicates are material at the beginning as the separator makes parsing differences. Internal duplicates are fine to normalize.

Reply:
Yes. This skips duplicates _after_ the beginning has been processed (that's what the line above is doing). The problem was that we weren't skipping duplicates immediately after what this code refers to as the "root" (i.e. after UNC paths or drive letters).